scripts/lib/upgrade-zulip-stage-2

```diff
@@ -99,6 +106,23 @@ def shutdown_server() -> None:
                           preexec_fn=su_to_zulip)
     IS_SERVER_UP = False
 
+# postgresql.version is required for database servers, but wasn't
+# previously; fill it in based on what the OS provides.
+if os.path.exists('/etc/init.d/postgresql'):
+    postgres_version = get_config(config_file, 'postgresql', 'version')
+    if not postgres_version:
+        default_postgres_version = {
+            ('debian', '10'): '11',
+            ('ubuntu', '18.04'): '10',
+            ('ubuntu', '20.04'): '12',
+            ('centos', '7'): '11',
+        }
+        if (vendor, os_version) in default_postgres_version:
+            postgres_version = default_postgres_version[(vendor, os_version)]
+        else:
+            error_desupported_os(vendor, os_version)
+        set_config(config_file, 'postgresql', 'version', postgres_version)
```

Comment:
I think this doesn’t work because nothing calls `config_file.write()`.

Reply:
Sigh, want to fix that while you're setup to test it?

Reply:
Gah.  Thanks for your fix in 60b800b1ac946bbb7bb522273f4cafa007355dbf. :(